### DISCUSSION TITLE
Error when exporting AAB

### DISCUSSION THREAD

**Discussion by RoyTheReal** [Sketchware-Pro/Sketchware-Pro#931] [Q&A]
I tried doing it unsigned, signed, removing the library it gives me the error about, etc! I can't export my app as an aab.

java.util.zip.ZipException: duplicate entry:
root/META-INF/com.google.android.mater
ial_material.version
at
java.util.zip.ZipOutputStream.putNextEntry
(ZipOutputStream.java:234)
at
mod.jbk.build.compiler.bundle.AppBundle
Compiler.createModuleMainArchive(AppB
undleCompiler.java:224)
at
com.besome.sketch.export.ExportProject
Activity$BuildingA syncTask.b(ExportProje
ctActivity.java:776)
at a.a.a.MA.a(Unknown Source:11)
at
com.besome.sketch.export.ExportProject
Activity$BuildingAsyncTask.dolnBackgrou
nd(ExportProjectActivity.java:895)
at
com.besome.sketch.export.ExportProject
Activity$BuildingA syncTask.dolnBackgrou
nd(ExportProjectActivity.java:610)
at
android.os.AsyncTask$3.cal(AsyncTask.ja
va:394)
at
java.util.concurrent.FutureTask.run(Future
Task.java:264)
at
android.os. AsyncTask$SerialExecutor$1.r
un(AsyncTask.java:305)
at
java.util.concurrent.ThreadPoolExecutor.ru
nWorker(ThreadPoolExecutor.java:1137)
at
java.util.concurrent.ThreadPoolExecutor$
Worker.run(ThreadPoolExecutor.java:637)
at
java.lang.Thread.run(Thread.java:1012)

**Comment by j-px** [comment]
did you read the error message? it's complaining about duplicated files because you're adding the material library on your own while not disabling AppCompat or having the material library excluded in Exclude built-in libraries.

**Reply by RoyTheReal** [reply]
I'm sorry sir, I'm not super smart at understanding the craziness of the error messages. Can you please explain how I can fix this in more details?

**Reply by RoyTheReal** [reply]
Where is exclude built-in libraries? Also, I already tried disabling it and I got another error.

**Reply by RoyTheReal** [reply]
Also, when I disable AppCompat, I get an error message and it removes my FAB

**Reply by j-px** [reply]
Exclude built-in libraries is only available in nightly builds so far. You can get them from [here, they're built with GitHub Actions](https://github.com/Sketchware-Pro/Sketchware-Pro/actions?query=branch%3Amain+event%3Apush).

When you have a nightly build that contains it, it's available in Library Manager (not Local library Manager!). More info on it is available at #696.

**Reply by j-px** [reply]
a duplicated entry error means two libraries are in conflict. you must remove one or the other. since you probably want the material library added via Local library Manager, exclude the material library Sketchware Pro adds with Exclude built-in libraries.
